lib/at_cmd/at_cmd.c

```diff
@@ -182,6 +182,7 @@ static void socket_thread_fn(void *arg1, void *arg2, void *arg3)
 				if (response_buf_len > payload_len) {
 					memcpy(response_buf, item->data,
 					       payload_len);
+					response_buf[payload_len] = '\0';
```

Comment:
I am not sure why we need this line? The NULL terminator should already be there. Have you seen responses from the modem that is not NULL terminated?

Reply:
Hmm, it does seem like there should be. I see it's even explicitly done in get_return_code().

I'm seeing a buffer I am using to send AT commands getting filled in with what appears to be only the string (sans NULL byte), which results in the new response getting printed and then any old response that was longer getting its extra bytes printed out. Since get_return_code() should be NULL-terminating the string, there is likely something in my code that isn't working. I'll take another look in a little bit.